proto/proto/dex.proto

```diff
@@ -12,28 +12,66 @@ message Swap {
   MockFlowCiphertext enc_amount_1 = 2;
   // Encrypted amount of asset 2 of the trading pair.
   MockFlowCiphertext enc_amount_2 = 3;
+  // Encapsulates the authorized fields of the Swap transaction, used in signing.
+  SwapBody body = 4;
+  // The spend authorization signature is effecting data.
+  crypto.SpendAuthSignature auth_sig = 5;
+}
+
+// A transaction action that obtains assets previously confirmed
+// via a Swap transaction. Does not include a spend authorization
+// signature, as it is only capable of consuming the NFT from a
+// Swap transaction.
+message SwapClaim {
+  // Proves the spent note records 1 unit of the swap NFT.
+  bytes zkproof = 1;
+  // The nullifier for the Swap NFT to be consumed.
+  crypto.Nullifier nullifier = 2;
+  // The fee allows `SwapClaim` without an additional `Spend`.
+  transaction.Fee fee = 3;
+  // Note output for asset 1.
+  crypto.NotePayload output_1 = 4;
+  // Note output for asset 2.
+  crypto.NotePayload output_2 = 5;
+  // Root for the associated `Swap` action.
```

Comment:
This should probably clarify that it's the *block* root, identifying the block the `Swap` was included in.